bundles/models/models_service.go

```diff
@@ -85,6 +87,41 @@ func (ms *Service) GetModelProto(ctx context.Context, tx *gorm.DB, owner,
 func (ms *Service) ModelList(p *gz.PaginationRequest, tx *gorm.DB, owner *string,
 	order, search string, likedBy *users.User, user *users.User, categories *category.Categories) (*fuel.Models, *gz.PaginationResult, *gz.ErrMsg) {
 
+	// Get a boolean that indicates if this a basic `GET /models` or `GET /models?page=N` query.
+	// In this case, we can ideally use the memdory cache to reduce the
+	// DB burden.
+	// Note: the PerPage default value is 20.
+	basicQuery := owner == nil && order == "" && search == "" && likedBy == nil && p != nil && (!p.PageRequested || (p.PageRequested && p.PerPage == 20)) && tx.Value == nil
+
+	paginationCacheKey := "models_list_pagination"
+	modelsCacheKey := "models_list_models"
+	if p != nil && p.PageRequested && p.PerPage == 20 {
+		paginationCacheKey = fmt.Sprintf("%s%d", paginationCacheKey, p.Page)
+		modelsCacheKey += fmt.Sprintf("%s%d", modelsCacheKey, p.Page)
+	}
+
+	// Try to get the memory cached results if it's a basic query.
+	if basicQuery {
+		paginationItem, errPagination := globals.QueryCache.Get(paginationCacheKey)
+		modelsItem, errModels := globals.QueryCache.Get(modelsCacheKey)
+
+		// If no errors, then unmarshal the bytes to the structs.
+		// Otherwise the normal query will be performed
+		if errPagination == nil && errModels == nil {
+			paginationResult := &gz.PaginationResult{}
+			modelsResult := &fuel.Models{}
+
+			errPagination = json.Unmarshal(paginationItem.Value, paginationResult)
+			errModels = proto.Unmarshal(modelsItem.Value, modelsResult)
+
+			// If no errors, then return the result. Otherwise do the normal
+			// query.
+			if errPagination == nil && errModels == nil {
+				return modelsResult, paginationResult, nil
+			}
+		}
+	}
```

Comment:
Let's extract this block into a function.

Reply:
Done in [ca689c0](https://github.com/gazebo-web/fuel-server/pull/21/commits/ca689c01570d686f24b99050b213273719ec92fc)